### ISSUE TITLE
[Post-services] Add `author`, `title`, `publishDate` fields to '/' route

### ISSUE THREAD

**Issue by Kevan-Y** [Seneca-CDOT/telescope#3615]
Our current Post-services `/` route will return an array of posts see JSON below.
```json
[
  {
    "id": "33a8c63f31",
    "url": "https://api.telescope.cdot.systems/v1/posts/33a8c63f31"
  },
  {
    "id": "6f3d109b82",
    "url": "https://api.telescope.cdot.systems/v1/posts/6f3d109b82"
  },
  {
    "id": "3253659d90",
    "url": "https://api.telescope.cdot.systems/v1/posts/3253659d90"
  }
  ]
  ```
 Since our React native app design uses a click and reads design. When fetching the list of posts, we have to do another fetch to get `author`, `title`, `date`. 
 This is inefficient since it makes a lot of unnecessary requests, we only need to request post detail on click to a post.
![image](https://user-images.githubusercontent.com/58233223/169631100-399ba510-e50d-4fe2-a0f5-8fb44b4b4cc4.png)

The solution is to add `author`, `title`, and `publishDate` to the object.
Sample JSON:
```json
[
  {
    "id": "33a8c63f31",
    "url": "https://api.telescope.cdot.systems/v1/posts/33a8c63f31",
    "author":"",
    "title":"",
    "publishDate":""
  },
  {
    "id": "6f3d109b82",
    "url": "https://api.telescope.cdot.systems/v1/posts/6f3d109b82",
     "author":"",
    "title":"",
    "publishDate":""
  },
  {
    "id": "3253659d90",
    "url": "https://api.telescope.cdot.systems/v1/posts/3253659d90",
     "author":"",
    "title":"",
    "publishDate":""
  }
  ]
  ```

**Comment by humphd** [comment]
Probably adding some kind of URL param to return a fully hydrated object is best (i.e., don't do it unless requested).  For example:

- `GET /posts` returns `id` and `url` only (current behaviour)
- `GET /posts?expand=1` returns some version of what you suggest above

This way you can opt into the extra calls this will require.

**Comment by DukeManh** [comment]
> Probably adding some kind of URL param to return a fully hydrated object is best (i.e., don't do it unless requested). For example:
> 
>     * `GET /posts` returns `id` and `url` only (current behaviour)
> 
>     * `GET /posts?expand=1` returns some version of what you suggest above
> 
> 
> This way you can opt into the extra calls this will require.

I agree.

**Comment by liutng** [comment]
Hi, 

Can I have this? I've already fixed this issue.

Does this fixed version look right?
![image](https://user-images.githubusercontent.com/8223139/202293444-08851017-e02c-40ba-8dbc-422f86ccf7b2.png)

**Comment by Kevan-Y** [comment]
> Hi, 
> 
> 
> 
> Can I have this? I've already fixed this issue.
> 
> 
> 
> Does this fixed version look right?
> 
> ![image](https://user-images.githubusercontent.com/8223139/202293444-08851017-e02c-40ba-8dbc-422f86ccf7b2.png)
> 
> 
> 
> 
> 
> 
> 
> 

Welcome, great it seems what we want

**Comment by manekenpix** [comment]
Closed via #3771
